docs/Glossary.md

```diff
@@ -0,0 +1,141 @@
+
+The structure of an action object is really up to you.  Data classes are a good choice, since `toString()` is impemented and can be useful for logging. Data inside the action is used to update the state inside reducers.  For example, an API request result may be inside an Action.
+
+See also [async action](#async-action) below.
+
+## Reducer
+
+```kotlin
+typealias Reducer<State> = (state: State, action: Any) -> State
+```
+
+A _reducer_ (also called a _reducing function_) is a function that accepts an accumulation and a value and returns a new accumulation. They are used to reduce a collection of values down to a single value.
+
+Reducers are not unique to Redux—they are a fundamental concept in functional programming. Even most non-functional languages, like Kotlin, have a built-in API for reducing. In Kotlin, reduce is a[`function on collections and many types`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/reduce.html).
+
+In Redux, the accumulated value is the state object, and the values being accumulated are actions. Reducers calculate a new state given the previous state and an action. They must be _pure functions_—functions that return the exact same output for given inputs. They should also be free of side-effects. This is what enables exciting features like hot reloading and time travel.
+
+Reducers are the most important concept in Redux.
+
+_Do not put API calls into reducers._
+
+## Dispatching Function
+
+```kotlin
+typealias Dispatcher = (Any) -> Any
+```
+
+A _dispatching function_ (or simply _dispatch function_) is a function that accepts an action or an [async action](#async-action); it then may or may not dispatch one or more actions to the store.
+
+We must distinguish between dispatching functions in general and the base [`dispatch`](api/Store.md#dispatchaction) function provided by the store instance without any middleware.
+
+The base dispatch function _always_ synchronously sends an action to the store's reducer, along with the previous state returned by the store, to calculate a new state. It expects actions to be plain objects ready to be consumed by the reducer.
+
+[Middleware](#middleware) wraps the base dispatch function. It allows the dispatch function to handle [async actions](#async-action) in addition to actions. Middleware may transform, delay, ignore, or otherwise interpret actions or async actions before passing them to the next middleware. See below for more information.
+
+## Action Creator
+
+Action Creators are used in JS Redux, but not as desirable in Kotlin.  They are simply a function that creates an Action or Async Action.  In Kotlin, since we are typically using Data classes this is not necessary.  Data class constructors provide a nice syntax for creating actions.  Using the Action Creator pattern in Kotlin works as well, and is really up to you as to how and where actions are created.
```

Comment:
```suggestion
Action Creators are used in JS Redux, but are not as desirable in Kotlin.  They are simply a function that creates an Action or Async Action.  In Kotlin, since we are typically using Data classes this is not necessary.  Data class constructors provide a nice syntax for creating actions.  Using the Action Creator pattern in Kotlin works as well, and is really up to you as to how and where actions are created.
```